### DISCUSSION TITLE
Authenticator does not support the request.

### DISCUSSION THREAD

**Discussion by DMC0048** [symfony/symfony#52465] [Q&A]
Hello,

I'm actually having some troubles with my LoginFormAuthenticator, in console logs: 
 Checking support on authenticator. authenticator="App\\Security\\LoginFormAuthenticator"
Authenticator does not support the request.

Checking support on authenticator. authenticator="Symfony\\Component\\Security\\Http\\Authenticator\\FormLoginAuthenticator"
Authenticator does not support the request.

Actually im trying to get an POST connection, getting instead a GET answer. Into my login.html.twig POST action its defined .

_DB connection seems okay, invalid credentials as answer when trying to log in._

ADD class LoginFormAuthenticator.php:

`
class LoginFormAuthenticator extends AbstractLoginFormAuthenticator
{
    use TargetPathTrait;

    public const LOGIN_ROUTE = 'app_login';

    public function __construct(private UrlGeneratorInterface $urlGenerator)
    {
    }

    public function authenticate(Request $request): Passport
    {
        $email = $request->request->get('email', '');

        $request->getSession()->set(SecurityRequestAttributes::LAST_USERNAME, $email);

        return new Passport(
            new UserBadge($email),
            new PasswordCredentials($request->request->get('password', '')),
            [
                new CsrfTokenBadge('authenticate', $request->request->get('_csrf_token')),            
            ]
        );
    }

    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
    {
        if ($targetPath = $this->getTargetPath($request->getSession(), $firewallName)) {
            return new RedirectResponse($targetPath);
        }

        // return new RedirectResponse($this->urlGenerator->generate('some_route'));
        return new RedirectResponse($this->urlGenerator->generate('app_dashboard'));
        //throw new \Exception('TODO: provide a valid redirect inside '.__FILE__);
    }
 
    protected function getLoginUrl(Request $request): string
    {
        return $this->urlGenerator->generate(self::LOGIN_ROUTE);
    }
}`

ADD class AbstractLoginFormAuthenticator -> function supports()

`    public function supports(Request $request): bool
    {
        return $request->isMethod('POST') && $this->getLoginUrl($request) === $request->getRequestUri();
    }`

**Comment by MatTheCat** [comment]
Hello,

I’m not sure to understand your issue. In the end, is the user logged in or not?

It seems like you make no difference between the request logging the user and the one after being redirected. You can see in `LoginFormAuthenticator::onAuthenticationSuccess` the user will be redirected to your `app_dashboard` route after being logged in, so this will make two different requests:

1. logging the user, which is a POST
2. displaying the dashboard after being redirected, which is a GET

If your dashboard is behind the same firewall than your `LoginFormAuthenticator` then it will try to authenticate the request and log that it does not support it, which is expected.
